### PR TITLE
Rename SimPSFToolkit.py module

### PR DESCRIPTION
**Pull request by nadamoukaddem**
solves #112 
Changes Made:

- Renamed `SimPSFToolkit.py` to`psf_simulator.py`

- Renamed the class to `PSFSimulator`

- Updated all package and class imports throughout the codebase

### CODE REVIEW

File: docs/source/wf_psf.sims.SimPSFToolkit.rst
Comment (jeipollack) [inline docs/source/wf_psf.sims.SimPSFToolkit.rst:1]: The module name is psf_simulator.

**Review by jeipollack — COMMENTED** [review]
I am checking the first few changes and noticed that you didn't change references and imports of the new module name: `psf_simulator` instead calling it PSFSimulator. PSFSimulator is the name of the class not the module. 

Can you please go through and make sure module imports, class imports and doc strings are correct?

Make sure you tested that everything works as expected.

File: src/wf_psf/psf_models/tf_psf_field.py
Comment (jeipollack) [inline src/wf_psf/psf_models/tf_psf_field.py:1]: All of the doc strings in this module are incorrect (even the old version was wrong).

It should be `wf_psf.sims.psf_simulator.PSFSimulator`.  Do understand why?

File: src/wf_psf/metrics/metrics.py [hunk 1]
```diff
@@ -640,7 +640,7 @@ def gen_GT_wf_model(test_wf_file_path, pred_output_Q=1, pred_output_dim=64):
 
     ## Generate initializations
     # Prepare np input
-    simPSF_np = SimPSFToolkit(
+    simPSF_np = PSFSimulator(
```
Comment (jeipollack) [inline src/wf_psf/metrics/metrics.py]: This should be `psf_simulator.PSFSimulator`

File: src/wf_psf/data/training_preprocessing.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 import wf_psf.utils.utils as utils
 import tensorflow as tf
 import tensorflow_addons as tfa
-import wf_psf.sims.SimPSFToolkit as SimPSFToolkit
+import wf_psf.sims.psf_simulator as psf_simulator
```
Comment (jeipollack) [inline src/wf_psf/data/training_preprocessing.py]: This module import is not used, so let's remove it.

File: src/wf_psf/psf_models/psf_model_parametric.py [hunk 3]
```diff
@@ -170,7 +170,7 @@ def predict_mono_psfs(self, input_positions, lambda_obs, phase_N):
 
         phase_N: int
             Required wavefront dimension. Should be calculated with as:
-            ``simPSF_np = wf.SimPSFToolkit(...)``
+            ``simPSF_np = wf.PSFSimulator(...)``
```
Comment (jeipollack) [inline src/wf_psf/psf_models/psf_model_parametric.py]: change to: `wf_psf.sims.psf_simulator.PSFSimulator`

File: src/wf_psf/psf_models/psf_model_semiparametric.py [hunk 4]
```diff
@@ -223,7 +223,7 @@ def predict_mono_psfs(self, input_positions, lambda_obs, phase_N):
 
         phase_N: int
             Required wavefront dimension. Should be calculated with as:
-            ``simPSF_np = wf.SimPSFToolkit(...)``
+            ``simPSF_np = wf.PSFSimulator(...)``
```
Comment (jeipollack) [inline src/wf_psf/psf_models/psf_model_semiparametric.py]: change to: `wf_psf.sims.psf_simulator.PSFSimulator`

**Review by jeipollack — CHANGES_REQUESTED** [review]
There are a few more changes to implement.

**Comment by nadamoukaddem** [comment]
> There are a few more changes to implement.

Is it better to make all changes in one commit, or to commit each file separately?

**Comment by jeipollack** [comment]
If you're performing exactly the same change for several files you can do it in one commit.  If it's a different type of change then it should go in another commit.

Also, please be sure to review the Workflow which took effect especially when it comes to branch naming conventions (in particular for next time).

File: src/wf_psf/psf_models/tf_psf_field.py
Comment (nadamoukaddem) [reply src/wf_psf/psf_models/tf_psf_field.py:1]: because we want to call the class that will give us wavefront information ?

Comment (jeipollack) [reply src/wf_psf/psf_models/tf_psf_field.py:1]: I meant do you know what `wf_psf.sims.psf_simulator.PSFSimulator` means?

Comment (nadamoukaddem) [reply src/wf_psf/psf_models/tf_psf_field.py:1]: that we're accessing the`PSFSimulator` class.

**Comment by nadamoukaddem** [comment]
Okay, thank you, Jennifer. 
I'll make sure to follow the branch naming conventions next time. The conventions include three cases: Feature, Bug, and Hotfix. For my case, it will be 'enhancement/issue-number/short-description'?

File: src/wf_psf/psf_models/tf_psf_field.py
Comment (jeipollack) [reply src/wf_psf/psf_models/tf_psf_field.py:1]: yes, but what is everything before it? 
How do you import something (e.g. class, function, variable) from one module into another?

**Comment by jeipollack** [comment]
> Okay, thank you, Jennifer.
> I'll make sure to follow the branch naming conventions next time. The conventions include three cases: Feature, Bug, and Hotfix. For my case, it will be 'enhancement/issue-number/short-description'?

No, the names must be from the three categories.  Which one would it be then?

File: src/wf_psf/psf_models/tf_psf_field.py
Comment (nadamoukaddem) [reply src/wf_psf/psf_models/tf_psf_field.py:1]: `psf-simulator` is the module or submodule, `sims` is a module and `wf_psf` is the package

**Comment by nadamoukaddem** [comment]
> > Okay, thank you, Jennifer.
> > I'll make sure to follow the branch naming conventions next time. The conventions include three cases: Feature, Bug, and Hotfix. For my case, it will be 'enhancement/issue-number/short-description'?
> 
> No, the names must be from the three categories. Which one would it be then?

It's not a new feature, so I would say it's more like a bug.

File: src/wf_psf/psf_models/tf_psf_field.py
Comment (jeipollack) [reply src/wf_psf/psf_models/tf_psf_field.py:1]: you have a typo: `psf_simulator` is a module.
`sims` is a (sub)-package
`wf_psf` is a package.

The import statement is an absolute import, where the full path has been specified.

**Comment by nadamoukaddem** [comment]
> There are a few more changes to implement.

Changes have been implemented.

**Comment by jeipollack** [comment]
Thanks for the update. Did you test with runs of training and metrics to ensure no breaking behaviour?

**Comment by nadamoukaddem** [comment]
> Thanks for the update. Did you test with runs of training and metrics to ensure no breaking behaviour?

Yes, but there are tests that require a GPU and they are skipped.

**Comment by jeipollack** [comment]
Not unit tests. I meant validation tests. Launch a batch script and make sure everything runs with no errors.

**Comment by nadamoukaddem** [comment]
> Not unit tests. I meant validation tests. Launch a batch script and make sure everything runs with no errors.

Okay, I will try testing it with Google Colab because Feynman is undergoing maintenance.

**Comment by nadamoukaddem** [comment]
> Not unit tests. I meant validation tests. Launch a batch script and make sure everything runs with no errors.

I tested the code with a few epochs and got the expected output without any errors.

**Comment by jeipollack** [comment]
Can you share your validation reports?

**Comment by nadamoukaddem** [comment]
[wf-psf_renameModule.log](https://github.com/CosmoStat/wf-psf/files/14604421/wf-psf_renameModule.log)

**Comment by jeipollack** [comment]
You uploaded a single log file of a run. I want to make sure you understand what is meant by validation reports that should be provided.  Can you explain more what this run is? What are the outputs you expected?

**Comment by nadamoukaddem** [comment]
> You uploaded a single log file of a run. I want to make sure you understand what is meant by validation reports that should be provided. Can you explain more what this run is? What are the outputs you expected?

After renaming the module, I ran WaveDiff. If everything is working correctly, the code should run without errors, and the metrics should have similar logical values as before renaming. The values in the log seem reasonable, but for full assurance, we should compare them to the log from before the renaming.

**Comment by jeipollack** [comment]
Thanks for this answer. Just a clarification, if you are using the same random seed and configuration for before-the-changes and after-the-changes run the results should be **exact**, not similar.  If they are not exact, it is important to investigate why they are not the same.

**Comment by nadamoukaddem** [comment]
Yes, by 'similar' I meant the same. As you mentioned, if we are using the same random seed. Thanks for the specifics.

**Comment by jeipollack** [comment]
Similar /= Same. 

Are you going to correct your report, so it's clear which run is the `before-changes` and `after-changes` runs? If you don't or can't run the validation tests, then you need to do the two runs and provide the corresponding logs.

**Comment by nadamoukaddem** [comment]
the log I provided is after-changes run

**Comment by jeipollack** [comment]
It is difficult as a reviewer to read bits of information across multiple comments. According to the Pull Request template, assignees should provide a summary of their checks in the original post. Thus, it's possible for you to edit it.  Again provide a complete report of the validation tests.  

Send me a notification when you're done and ready for me to review the Pull Request.

**Comment by nadamoukaddem** [comment]
[metrics_after_changes.log](https://github.com/CosmoStat/wf-psf/files/14639773/metrics_after_changes.log)
[metrics_before_changes.log](https://github.com/CosmoStat/wf-psf/files/14639774/metrics_before_changes.log)
Done.

File: src/wf_psf/data/training_preprocessing.py (hunk 2 above)
Comment (jeipollack) [reply src/wf_psf/data/training_preprocessing.py]: I stated two week ago that this module import is not used in this module. I asked you to remove it. You did not do yet.

**Review by jeipollack — APPROVED** [review]
Thanks for working on this.

**Pull request merged**